msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -979,64 +988,51 @@ protected void validateAcquireTokenSilentParameters(AcquireTokenSilentParameters
 
     @Override
     public void acquireToken(@NonNull final AcquireTokenParameters acquireTokenParameters) {
-        acquireTokenParameters.setAccountRecord(
-                getAccountRecord(
-                        acquireTokenParameters.getAccount(),
-                        getRequestTenantId(
-                                mPublicClientConfiguration,
-                                acquireTokenParameters
-                        )
-                )
-        );
-
-        validateAcquireTokenParameters(acquireTokenParameters);
-
-        final AcquireTokenOperationParameters params = OperationParametersAdapter.
-                createAcquireTokenOperationParameters(
-                        acquireTokenParameters,
-                        mPublicClientConfiguration
-                );
+        sBackgroundExecutor.submit(new Runnable() {
```

Comment:
Isn't beginInteractive already dispatching to a bg thread?

Reply:
It does inside of the ApiDispatcher, but at this entry point we're still on the main thread. To support querying OpenId metadata I'm moving this to a bg thread _earlier_

Reply:
ooooh.. okay

Reply:
In that case can you add a comment here?

Reply:
alternatively you could pass this whole operation as a callback to selectAccountRecordForTokenRequest. not sure which one is uglier though :D

Reply:
Hahah ok - I think I'll leave as is. I'll take your suggestion and add a comment 👍

Reply:
Resolved in 82cfd1150f97267050368f44cc3959204b98fe10